packages/resolve-redux/src/subscribe_adapter.js

```diff
@@ -13,9 +13,24 @@ export default function subscribeAdapter() {
     transports: ['websocket']
   })
 
+  let resolveClientId = null
+  let clientIdPromise = new Promise(resolve => (resolveClientId = resolve))
+
   socket.on('event', event => onEvent(JSON.parse(event)))
 
-  socket.on('disconnect', reason => onDisconnect(reason))
+  socket.on('connect', () => resolveClientId(socket.id))
+
+  socket.on('reconnect', () => {
+    resolveClientId = null
+    clientIdPromise = new Promise(resolve => (resolveClientId = resolve))
```

Comment:
Copy-paste

Reply:
Not exactly from POV late-resolving promises from their activation contexts. May be encapsulated in late-resolved promise construction, but some "copy-paste" like code will remained, since promise object and it's late resolver should be recharged